packages/radix-ui-themes/src/helpers/breakpoints.ts

```diff
@@ -67,11 +71,151 @@ function withBreakpoints(
   return classes.join(' ');
 }
 
+function getResponsiveStyles({
+  allowAribtraryValues,
+  className,
+  customProperty,
+  ...args
+}: GetResponsiveCustomPropertiesOptions & GetResponsiveClassNamesOptions) {
+  const classNames = getResponsiveClassNames({
+    allowAribtraryValues,
+    className,
+    ...args,
+  });
+
+  if (allowAribtraryValues) {
+    const customProperties = getResponsiveCustomProperties({ customProperty, ...args });
+    return [classNames, customProperties] as const;
+  } else {
+    return [classNames, undefined] as const;
+  }
+}
+
+interface GetResponsiveClassNamesOptions {
+  className?: string;
+  value: ResponsiveWide<string> | Responsive<string> | undefined;
+  values?: string[] | readonly string[];
+  allowAribtraryValues?: boolean;
+  map?: (value: string | undefined) => string | undefined;
+}
+
+function getResponsiveClassNames({
```

Comment:
This function will ultimately replace `withBreakpoints`. It basically does the same thing, but accounts for creating corresponding classnames when an arbitrary value is passed.

It will need to work with boolean props first though.